examples/golang/components/http-client-tinygo/hello.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"log"
+	"math"
+
+	http "github.com/wasmcloud/wasmcloud/examples/golang/components/http-client-tinygo/gen"
+)
+
+// Helper type aliases to make code more readable
+type HttpRequest = http.ExportsWasiHttp0_2_0_IncomingHandlerIncomingRequest
+type HttpResponseWriter = http.ExportsWasiHttp0_2_0_IncomingHandlerResponseOutparam
+type HttpOutgoingResponse = http.WasiHttp0_2_0_TypesOutgoingResponse
+type HttpError = http.WasiHttp0_2_0_TypesErrorCode
+
+type HttpServer struct{}
+
+func init() {
+	httpserver := HttpServer{}
+	// Set the incoming handler struct to HttpServer
+	http.SetExportsWasiHttp0_2_0_IncomingHandler(httpserver)
+}
+
+func (h HttpServer) Handle(request HttpRequest, responseWriter HttpResponseWriter) {
+	req := http.NewOutgoingRequest(http.NewFields())
+	req.SetScheme(http.Some[http.WasiHttp0_2_0_TypesScheme](http.WasiHttp0_2_0_TypesSchemeHttps()))
+	req.SetAuthority(http.Some("postman-echo.com"))
+	req.SetPathWithQuery(http.Some("/bytes/100/mb"))
+	result := http.WasiHttp0_2_0_OutgoingHandlerHandle(req, http.None[http.WasiHttp0_2_0_OutgoingHandlerRequestOptions]())
+	if result.IsOk() {
+		incomingResponse := result.Unwrap()
+		incomingResponse.Subscribe().Block()
+		resp := incomingResponse.Get().Unwrap().Unwrap().Unwrap()
+		if resp.Status() == 200 {
+			responseBody := resp.Consume().Unwrap()
+			headers := http.NewFields()
+			httpResponse := http.NewOutgoingResponse(headers)
+			httpResponse.SetStatusCode(200)
+			body := httpResponse.Body().Unwrap()
+			bodyWrite := body.Write().Unwrap()
+			instream := responseBody.Stream().Unwrap()
+			okResponse := http.Ok[HttpOutgoingResponse, HttpError](httpResponse)
+			http.StaticResponseOutparamSet(responseWriter, okResponse)
+			splice(instream, bodyWrite)
+			bodyWrite.Drop()
+			http.StaticOutgoingBodyFinish(body, http.None[http.WasiHttp0_2_0_TypesTrailers]())
+		}
```

Comment:
I think here what might be better is set the outgoing response to the returned status code and reading the body even if the status code isn't 200, what do you think?

Reply:
so we want to return the body as is ? I can just remove the status code check in that case .

Reply:
I have made the changes as requested. Please review